src/ncp/ncp_base_dispatcher.cpp

```diff
@@ -37,1028 +37,532 @@ namespace Ncp {
+      {SPINEL_PROP_NET_NETWORK_NAME, &NcpBase::HandlePropertySet<SPINEL_PROP_NET_NETWORK_NAME>},
+      {SPINEL_PROP_NET_XPANID, &NcpBase::HandlePropertySet<SPINEL_PROP_NET_XPANID>},
+      {SPINEL_PROP_NET_MASTER_KEY, &NcpBase::HandlePropertySet<SPINEL_PROP_NET_MASTER_KEY>},
+      {SPINEL_PROP_NET_KEY_SEQUENCE_COUNTER, &NcpBase::HandlePropertySet<SPINEL_PROP_NET_KEY_SEQUENCE_COUNTER>},
+#if OPENTHREAD_FTD
+      {SPINEL_PROP_NET_PARTITION_ID, &NcpBase::HandlePropertySet<SPINEL_PROP_NET_PARTITION_ID>},
 #endif
-#if OPENTHREAD_CONFIG_LEGACY_ENABLE
-    case SPINEL_PROP_NEST_LEGACY_ULA_PREFIX:
-        handler = &NcpBase::HandlePropertySet<SPINEL_PROP_NEST_LEGACY_ULA_PREFIX>;
-        break;
+      {SPINEL_PROP_NET_REQUIRE_JOIN_EXISTING, &NcpBase::HandlePropertySet<SPINEL_PROP_NET_REQUIRE_JOIN_EXISTING>},
+      {SPINEL_PROP_NET_KEY_SWITCH_GUARDTIME, &NcpBase::HandlePropertySet<SPINEL_PROP_NET_KEY_SWITCH_GUARDTIME>},
+#if OPENTHREAD_FTD
+      {SPINEL_PROP_NET_PSKC, &NcpBase::HandlePropertySet<SPINEL_PROP_NET_PSKC>},
+      {SPINEL_PROP_THREAD_LOCAL_LEADER_WEIGHT, &NcpBase::HandlePropertySet<SPINEL_PROP_THREAD_LOCAL_LEADER_WEIGHT>},
 #endif
-    case SPINEL_PROP_CNTR_RESET:
-        handler = &NcpBase::HandlePropertySet<SPINEL_PROP_CNTR_RESET>;
-        break;
-    case SPINEL_PROP_CNTR_ALL_MAC_COUNTERS:
-        handler = &NcpBase::HandlePropertySet<SPINEL_PROP_CNTR_ALL_MAC_COUNTERS>;
-        break;
-    case SPINEL_PROP_CNTR_MLE_COUNTERS:
-        handler = &NcpBase::HandlePropertySet<SPINEL_PROP_CNTR_MLE_COUNTERS>;
-        break;
-    case SPINEL_PROP_CNTR_ALL_IP_COUNTERS:
-        handler = &NcpBase::HandlePropertySet<SPINEL_PROP_CNTR_ALL_IP_COUNTERS>;
-        break;
-#if OPENTHREAD_CONFIG_CHILD_SUPERVISION_ENABLE
-    case SPINEL_PROP_CHILD_SUPERVISION_CHECK_TIMEOUT:
-        handler = &NcpBase::HandlePropertySet<SPINEL_PROP_CHILD_SUPERVISION_CHECK_TIMEOUT>;
-        break;
+      {SPINEL_PROP_THREAD_ASSISTING_PORTS, &NcpBase::HandlePropertySet<SPINEL_PROP_THREAD_ASSISTING_PORTS>},
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE
+      {SPINEL_PROP_THREAD_ALLOW_LOCAL_NET_DATA_CHANGE,
+       &NcpBase::HandlePropertySet<SPINEL_PROP_THREAD_ALLOW_LOCAL_NET_DATA_CHANGE>},
+#endif
+      {SPINEL_PROP_THREAD_MODE, &NcpBase::HandlePropertySet<SPINEL_PROP_THREAD_MODE>},
+      {SPINEL_PROP_IPV6_ML_PREFIX, &NcpBase::HandlePropertySet<SPINEL_PROP_IPV6_ML_PREFIX>},
+      {SPINEL_PROP_IPV6_ICMP_PING_OFFLOAD, &NcpBase::HandlePropertySet<SPINEL_PROP_IPV6_ICMP_PING_OFFLOAD>},
+      {SPINEL_PROP_IPV6_ICMP_PING_OFFLOAD_MODE, &NcpBase::HandlePropertySet<SPINEL_PROP_IPV6_ICMP_PING_OFFLOAD_MODE>},
+      {SPINEL_PROP_STREAM_NET, &NcpBase::HandlePropertySet<SPINEL_PROP_STREAM_NET>},
+      {SPINEL_PROP_STREAM_NET_INSECURE, &NcpBase::HandlePropertySet<SPINEL_PROP_STREAM_NET_INSECURE>},
+#if OPENTHREAD_CONFIG_JOINER_ENABLE
+      {SPINEL_PROP_MESHCOP_JOINER_COMMISSIONING, &NcpBase::HandlePropertySet<SPINEL_PROP_MESHC
```

Comment:
Looks like the array model is only used for Get/Set property handlers.
Should we change Insert/Remove handlers  as well? (I think it's better to have the same model all prop handlers).

Reply:
There are only a few handlers for insert and remove, so I intended leave them as is. However, I am fine if someone else want to change them to binary search. I make this PR higher priority because it is blocking our GitHub sync process.